core/backend/src/ECDb.ts

```diff
@@ -283,244 +281,91 @@ export class ECDb implements IDisposable {
-          try {
-            const timeElapsedSinceLastActivity = Date.now() - stats.lastActivityTime;
-            if (timeElapsedSinceLastActivity < logIntervalMs) {
-              const statistics = JSON.parse(this.nativeDb.captureConcurrentQueryStats(false));
-              Logger.logInfo(loggerCategory, "Concurrent query statistics", () => statistics);
-            }
-          } catch { }
-        } else {
-          clearInterval(stats.logTimerHandle);
-          stats.logTimerHandle = null;
-        }
-      }, logIntervalMs);
-    }
-
-    if (!bindings) bindings = [];
-    if (!limit) limit = {};
-    if (!quota) quota = {};
-    if (!priority) priority = QueryPriority.Normal;
-
-    return new Promise<QueryResponse>((resolve) => {
-      let sessionRestartToken = restartToken ? restartToken.trim() : "";
-      if (sessionRestartToken !== "")
-        sessionRestartToken = `${IModelHost.sessionId}:${sessionRestartToken}`;
-
-      const postResult = this.nativeDb.postConcurrentQuery(ecsql, JSON.stringify(bindings, Base64EncodedString.replacer), limit!, quota!, priority!, sessionRestartToken, abbreviateBlobs);
-      if (postResult.status !== IModelJsNative.ConcurrentQuery.PostStatus.Done)
-        resolve({ status: QueryResponseStatus.PostError, rows: [] });
-
-      const poll = () => {
-        if (!this.nativeDb || !this.nativeDb.isOpen()) {
-          resolve({ status: QueryResponseStatus.Done, rows: [] });
-        } else {
-          const pollResult = this.nativeDb.pollConcurrentQuery(postResult.taskId);
-          if (pollResult.status === IModelJsNative.ConcurrentQuery.PollStatus.Done) {
-            resolve({ status: QueryResponseStatus.Done, rows: JSON.parse(pollResult.result, Base64EncodedString.reviver) });
-          } else if (pollResult.status === IModelJsNative.ConcurrentQuery.PollStatus.Partial) {
-            const returnBeforeStep = pollResult.result.length === 0;
-            resolve({ status: QueryResponseStatus.Partial, rows: returnBeforeStep ? [] : JSON.parse(pollResult.result, Base64EncodedString.reviver) });
-          } else if (pollResult.status === IModelJsNative.ConcurrentQuery.PollStatus.Timeout)
-            resolve({ status: QueryResponseStatus.Timeout, rows: [] });
-          else if (pollResult.status === IModelJsNative.ConcurrentQuery.PollStatus.Pending)
-            setTimeout(() => { poll(); }, IModelHost.configuration!.concurrentQuery.pollInterval);
-          else if (pollResult.status === IModelJsNative.ConcurrentQuery.PollStatus.Cancelled)
-            resolve({ status: QueryResponseStatus.Cancelled, rows: [pollResult.result] });
-          else
-            resolve({ status: QueryResponseStatus.Error, rows: [pollResult.result] });
-        }
-      };
-      setTimeout(() => { poll(); }, IModelHost.configuration!.concurrentQuery.pollInterval);
-    });
+  public async * query(ecsql: string, params?: QueryBinder, rowFormat?: QueryRowFormat, config?: QueryOptions): AsyncIterableIterator<any> {
```

Comment:
Here and elsewhere, replace `rowFormat?: QueryRowFormat` with `rowFormat: QueryRowFormat = QueryRowFormat.UseArrayIndexes` to make it clear at the call site what the default is.
`QueryRowFormat.Default` is unnecessary.